add TestKcpServe, demonstrate and enable kcp based rcp with encryption.

glycerine opened the pull request:
Add TestKcpServe; Demonstrate and enable encrypted rpc using kcp.

**Please answer these questions before sending pull requests. Thanks!**

### did you run tests in local？
Yes.

If there's a better way to add encryption for kcp based transport, let me know.

smallnest commented on server.go:
```diff
@@ -226,10 +231,12 @@ func validIP4(ipAddress string) bool {
+// It is blocked until receiving connectings from clients.
+// The blockCrypt argument can be nil if the caller
+// wishes to skip encryption, or is not using the kcp transport.
+func (s *Server) Serve(network, address string, blockCrypt kcp.BlockCrypt) (err error) {
```
kcp is a plugin and few uses will use it in production. It should not be a parameter here.

glycerine replied:
so maybe separate functions just for KCP?

Like: ServerKCP(), and StartKCP() ?

glycerine replied:
I'll add a KCPConfig element to Server.

smallnest commented on client.go:
```diff
@@ -292,10 +292,13 @@ func (c *Client) wrapCall(client *core.Client, ctx context.Context, serviceMetho
 //The done channel will signal when the call is complete by returning the same Call object. If done is nil, Go will allocate a new channel. If non-nil, done must be buffered or Go will deliberately crash.
-func (c *Client) Go(ctx context.Context, serviceMethod string, args interface{}, reply interface{}, done chan *core.Call) *core.Call {
-	rpcClient, _ := c.ClientSelector.Select(c.ClientCodecFunc)
+func (c *Client) Go(ctx context.Context, serviceMethod string, args interface{}, reply interface{}, done chan *core.Call) (*core.Call, error) {
```
should not change method because it be used in many products.

glycerine replied:
There is error information that is being lost here with the _ underscore, if the connection is mis-configured (this bit me when trying to figure out the kcp use). If the error is not returned, the alternative is to panic... which is also undesirable in production, but panic is probably equivalent to what happens now, only it would come earlier so would offer better insight into the problem. I'll change it to panic.

glycerine commented:
> If there's a better way to add encryption for kcp based transport, let me know.

Yes, I agree with your observations and concern about backwards compatibility. I asked about a better way above because I couldn't see one. How would you suggest using encrypted kcp with rpcx?

glycerine commented:
I've refactored to leave the interfaces the same to satisfy backwards compatibility.

I added a KCPConfig element to the Server instead.

Please take a look and let me know what you think.

Jason

glycerine commented:
The TravisCI failure happens on master branch too, so it appears unrelated to this PR.

smallnest commented:
merged. Thanks
